In [1]:
import pandas as pd
import numpy as np

total = 2985217

## Load Data

In [2]:
train_2016 = pd.read_csv("train_2016_v2.csv", parse_dates=["transactiondate"])
print(train_2016.shape)
train_2016.tail()

(90275, 3)


parcelid  logerror transactiondate
90270  10774160   -0.0356      2016-12-30
90271  12046695    0.0070      2016-12-30
90272  12995401   -0.2679      2016-12-30
90273  11402105    0.0602      2016-12-30
90274  12566293    0.4207      2016-12-30

In [3]:
property_2016 = pd.read_csv("properties_2016.csv")
print(property_2016.shape)
property_2016.tail()

/usr/local/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (22,32,34,49,55) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


(2985217, 58)


parcelid  airconditioningtypeid  architecturalstyletypeid  \
2985212  168176230                    NaN                       NaN   
2985213   14273630                    NaN                       NaN   
2985214  168040630                    NaN                       NaN   
2985215  168040830                    NaN                       NaN   
2985216  168040430                    NaN                       NaN   

         basementsqft  bathroomcnt  bedroomcnt  buildingclasstypeid  \
2985212           NaN          NaN         NaN                  NaN   
2985213           NaN          NaN         NaN                  NaN   
2985214           NaN          NaN         NaN                  NaN   
2985215           NaN          NaN         NaN                  NaN   
2985216           NaN          NaN         NaN                  NaN   

         buildingqualitytypeid  calculatedbathnbr  decktypeid  \
2985212                    NaN                NaN         NaN   
2985213                    NaN                NaN         NaN   
2985214                    NaN                NaN         NaN   
2985215                    NaN                NaN         NaN   
2985216                    NaN                NaN         NaN   

                ...           numberofstories  fireplaceflag  \
2985212         ...                       NaN            NaN   
2985213         ...                       NaN            NaN   
2985214         ...                       NaN            NaN   
2985215         ...                       NaN            NaN   
2985216         ...                       NaN            NaN   

         structuretaxvaluedollarcnt  taxvaluedollarcnt  assessmentyear  \
2985212                         NaN                NaN             NaN   
2985213                         NaN                NaN             NaN   
2985214                         NaN                NaN             NaN   
2985215                         NaN                NaN             NaN   
2985216                         NaN                NaN             NaN   

         landtaxvaluedollarcnt  taxamount  taxdelinquencyflag  \
2985212                    NaN        NaN                 NaN   
2985213                    NaN        NaN                 NaN   
2985214                    NaN        NaN                 NaN   
2985215                    NaN        NaN                 NaN   
2985216                    NaN        NaN                 NaN   

         taxdelinquencyyear  censustractandblock  
2985212                 NaN                  NaN  
2985213                 NaN                  NaN  
2985214                 NaN                  NaN  
2985215                 NaN                  NaN  
2985216                 NaN                  NaN  

[5 rows x 58 columns]

## Dictionary

### airconditioningtypeid
- Type of cooling system present in the home (if any)
- category / 7
- rate of NaN : 2173698 / total ( 72.8% ) 

In [4]:
print(len(property_2016["airconditioningtypeid"].value_counts()))
property_2016["airconditioningtypeid"].value_counts().head()

7


1.0     742364
13.0     58457
5.0       8795
11.0      1818
12.0        59
Name: airconditioningtypeid, dtype: int64

In [5]:
property_2016["airconditioningtypeid"].isna().sum()/ total

0.7281541006901676

### architecturalstyletypeid
- Architectural style of the home (i.e. ranch, colonial, split-level, etc…)
- category / 6-type
- 7.0, 8.0, 2.0, 21.0, 3.0, 5.0, 27.0, 10.0
- NaN : 2979156 / 2985217 ( 99.76% )

In [6]:
print(len(property_2016["architecturalstyletypeid"].value_counts()))
property_2016["architecturalstyletypeid"].value_counts().head()

8


7.0     5251
8.0      380
2.0      201
21.0     150
3.0       58
Name: architecturalstyletypeid, dtype: int64

In [7]:
property_2016["architecturalstyletypeid"].isna().sum()

2979156

### basementsqft
- Finished living area below or partially below ground level
- (scalar/category) / 751
- NaN : 2983589 / 2985217 ( 99.99% )

In [8]:
print(len(property_2016["basementsqft"].value_counts()))
property_2016["basementsqft"].value_counts().head()

751


1528.0    25
700.0     21
240.0     21
100.0     18
144.0     16
Name: basementsqft, dtype: int64

In [9]:
property_2016["basementsqft"].isna().sum()

2983589

### bathroomcnt
- Number of bathrooms in home including fractional bathrooms
- scalar / 37
- NaN : 11462 / 2985217 ( 0.3% )

In [10]:
print(len(property_2016["bathroomcnt"].value_counts()))
property_2016["bathroomcnt"].value_counts().head()

37


2.0    1218664
3.0     629351
1.0     499330
2.5     206856
4.0     132076
Name: bathroomcnt, dtype: int64

In [11]:
property_2016["bathroomcnt"].isna().sum()

11462

### bedroomcnt
- Number of bedrooms in home
- scalar / 21 
- NaN  : 11450 / total ( 0.3% )

In [12]:
print(len(property_2016["bedroomcnt"].value_counts()))
property_2016["bedroomcnt"].value_counts().head()

21


3.0    1169462
4.0     727052
2.0     605367
5.0     181222
0.0     121466
Name: bedroomcnt, dtype: int64

In [13]:
property_2016["bedroomcnt"].isna().sum()

11450

### buildingclasstypeid
- The building framing type (steel frame, wood frame, concrete/brick) 
- category / 5-types
- NaN : 2972588 / total ( 99.57% ) 

In [14]:
print(len(property_2016["buildingclasstypeid"].value_counts()))
property_2016["buildingclasstypeid"].value_counts().head()

5


4.0    9265
3.0    3161
2.0      81
1.0      65
5.0      57
Name: buildingclasstypeid, dtype: int64

In [15]:
property_2016["buildingclasstypeid"].isna().sum()

2972588

### buildingqualitytypeid
- Overall assessment of condition of the building from best (lowest) to worst (highest)
- category / 12-types
- NaN : 1056729 / total ( 35% )

In [16]:
print(len(property_2016["buildingqualitytypeid"].value_counts()))
property_2016["buildingqualitytypeid"].value_counts().head()

12


7.0     1133238
4.0      692160
1.0       69467
10.0      39713
12.0       3692
Name: buildingqualitytypeid, dtype: int64

In [17]:
property_2016["buildingqualitytypeid"].isna().sum()

1046729

### calculatedbathnbr
- Number of bathrooms in home including fractional bathroom
- scalar / 34
- NaN : 128912 / total ( 4.3% )

In [18]:
print(len(property_2016["calculatedbathnbr"].value_counts()))
property_2016["calculatedbathnbr"].value_counts().head()

34


2.0    1218652
3.0     629351
1.0     499322
2.5     206563
4.0     132076
Name: calculatedbathnbr, dtype: int64

In [19]:
property_2016["calculatedbathnbr"].isna().sum()

128912

### decktypeid
- Type of deck (if any) present on parcel
- category / 1-type
- NaN : 2968121 / total ( 99.42% )

In [20]:
print(len(property_2016["decktypeid"].value_counts()))
property_2016["decktypeid"].value_counts().head()

1


66.0    17096
Name: decktypeid, dtype: int64

In [21]:
property_2016["decktypeid"].isna().sum()

2968121

### threequarterbathnbr
- Number of 3/4 bathrooms in house (shower + sink + toilet)
- scalar / 7개
- NaN : 2673586 / total ( 89% )

In [22]:
print(len(property_2016["threequarterbathnbr"].value_counts()))
property_2016["threequarterbathnbr"].value_counts().head()

7


1.0    308959
2.0      2338
3.0       261
4.0        46
5.0        16
Name: threequarterbathnbr, dtype: int64

In [23]:
property_2016["threequarterbathnbr"].isna().sum()

2673586

### finishedfloor1squarefeet
- Size of the finished living area on the first (entry) floor of the home
- scalar / 4604 
- NaN : 2782500 / total ( 93.20% )

In [24]:
print(len(property_2016["finishedfloor1squarefeet"].value_counts()))
property_2016["finishedfloor1squarefeet"].value_counts().head()

4604


1252.0    923
1233.0    864
1120.0    705
1226.0    616
1612.0    597
Name: finishedfloor1squarefeet, dtype: int64

In [25]:
property_2016["finishedfloor1squarefeet"].isna().sum() / total

0.9320930438222749

### calculatedfinishedsquarefeet
- Calculated total finished living area of the home
- scalar / 12587
- NaN : 55565 (1.8%)

In [26]:
print(len(property_2016["calculatedfinishedsquarefeet"].value_counts()))
property_2016["calculatedfinishedsquarefeet"].value_counts().head()

12587


1440.0    8946
1344.0    7203
1200.0    7098
960.0     6014
1120.0    5918
Name: calculatedfinishedsquarefeet, dtype: int64

In [27]:
property_2016["calculatedfinishedsquarefeet"].isna().sum() / total

0.018613387234495852

### finishedsquarefeet6
- Base unfinished and finished area
- scalar / 4251
- NaN : 2963216 / total ( 99.26% )

In [28]:
print(len(property_2016["finishedsquarefeet6"].value_counts()))
property_2016["finishedsquarefeet6"].value_counts().head()

4251


952.0     101
800.0      90
3820.0     88
720.0      74
600.0      68
Name: finishedsquarefeet6, dtype: int64

In [29]:
property_2016["finishedsquarefeet6"].isna().sum()

2963216

### finishedsquarefeet12
- Finished living area
- scalar / 10670
- NaN : 276033 / total ( 9.2% )

In [30]:
print(len(property_2016["finishedsquarefeet12"].value_counts()))
property_2016["finishedsquarefeet12"].value_counts().head()

10670


1440.0    6941
1200.0    6636
1344.0    6235
1120.0    5660
960.0     5611
Name: finishedsquarefeet12, dtype: int64

In [31]:
property_2016["finishedsquarefeet12"].isna().sum() / total

0.0924666448033761

### finishedsquarefeet13
- Perimeter  living area
- scalar / 250
- NaN : 2977545 / total ( 99.74% )

In [32]:
print(len(property_2016["finishedsquarefeet13"].value_counts()))
property_2016["finishedsquarefeet13"].value_counts().head()

250


1440.0    1571
1344.0     709
1248.0     453
1536.0     355
720.0      245
Name: finishedsquarefeet13, dtype: int64

In [33]:
property_2016["finishedsquarefeet13"].isna().sum() / total

0.9974300025760271

### finishedsquarefeet15
- Total area
- scalar / 8816
- NaN : 2794419 / total ( 93.6% )

In [34]:
print(len(property_2016["finishedsquarefeet15"].value_counts()))
property_2016["finishedsquarefeet15"].value_counts().head()

8816


1536.0    647
1728.0    429
1600.0    415
1440.0    413
1800.0    402
Name: finishedsquarefeet15, dtype: int64

In [35]:
property_2016["finishedsquarefeet15"].isna().sum() / total

0.9360857183916613

### finishedsquarefeet50
-  Size of the finished living area on the first (entry) floor of the home
- scalar / 4802
- rate of NaN : 93.2 %

In [36]:
print(len(property_2016["finishedsquarefeet50"].value_counts()))
property_2016["finishedsquarefeet50"].value_counts().head()

4802


1252.0    923
1233.0    864
1120.0    705
1226.0    616
1612.0    597
Name: finishedsquarefeet50, dtype: int64

In [37]:
property_2016["finishedsquarefeet50"].isna().sum() / total

0.9320930438222749

### fips
-  Federal Information Processing Standard code -  see https://en.wikipedia.org/wiki/FIPS_county_code for more details
- category / 3
- rate of NaN : 0.3%

In [38]:
print(len(property_2016["fips"].value_counts()))
property_2016["fips"].value_counts().head()

3


6037.0    2009362
6059.0     741565
6111.0     222853
Name: fips, dtype: int64

In [39]:
property_2016["fips"].isna().sum() / total

0.0038312122703307665

### fireplacecnt
-  Number of fireplaces in a home (if any)
- scalar / 9
- rate of NaN : 89.5%

In [40]:
print(len(property_2016["fireplacecnt"].value_counts()))
property_2016["fireplacecnt"].value_counts().head()

9


1.0    269643
2.0     34407
3.0      7696
4.0       710
5.0       126
Name: fireplacecnt, dtype: int64

In [41]:
property_2016["fireplacecnt"].isna().sum() / total

0.895271600021037

### fireplaceflag
- Is a fireplace present in this home 
- category / 2 (True / False)
- rate of NaN : 99.8%

In [42]:
print(len(property_2016["fireplaceflag"].value_counts()))
property_2016["fireplaceflag"].value_counts().head()

1


True    5163
Name: fireplaceflag, dtype: int64

In [43]:
property_2016["fireplaceflag"].isna().sum() / total

0.9982704774895761

### fullbathcnt
- Number of full bathrooms (sink, shower + bathtub, and toilet) present in home
- scalar / 20
- rate of NaN : 4.3%

In [44]:
print(len(property_2016["fullbathcnt"].value_counts()))
property_2016["fullbathcnt"].value_counts().head()

20


2.0    1425661
3.0     660139
1.0     544928
4.0     151004
5.0      43918
Name: fullbathcnt, dtype: int64

In [45]:
property_2016["fullbathcnt"].isna().sum() / total

0.0431834603648579

### garagecarcnt
- Total number of garages on the lot including an attached garage
- scalar / 24
- rate of NaN : 70%

In [46]:
print(len(property_2016["garagecarcnt"].value_counts()))
property_2016["garagecarcnt"].value_counts().head()

24


2.0    660462
1.0    177579
3.0     19634
0.0     14005
4.0      8495
Name: garagecarcnt, dtype: int64

In [47]:
property_2016["garagecarcnt"].isna().sum() / total

0.7041196670124819

### garagetotalsqft
- Total number of square feet of all garages on lot including an attached garage
- scalar / 2240
- rate of NaN : 70%

In [48]:
print(len(property_2016["garagetotalsqft"].value_counts()))
property_2016["garagetotalsqft"].value_counts().head()

2240


0.0      196752
400.0     33443
360.0     29703
441.0     26753
462.0     25476
Name: garagetotalsqft, dtype: int64

In [49]:
property_2016["garagetotalsqft"].isna().sum() / total

0.7041196670124819

### hashottuborspa
- Does the home have a hot tub or spa
- category / 2 (True/False)
- rate of NaN : 97%

In [50]:
print(len(property_2016["hashottuborspa"].value_counts()))
property_2016["hashottuborspa"].value_counts().head()

1


True    69014
Name: hashottuborspa, dtype: int64

In [51]:
property_2016["hashottuborspa"].isna().sum() / total

0.9768814126410241

### heatingorsystemtypeid
- Type of home heating system
- category / 14
- rate of NaN : 39.4%

In [52]:
print(len(property_2016["heatingorsystemtypeid"].value_counts()))
property_2016["heatingorsystemtypeid"].value_counts().head()

14


2.0     1156830
7.0      595453
6.0       27480
24.0      21107
20.0       3216
Name: heatingorsystemtypeid, dtype: int64

In [53]:
property_2016["heatingorsystemtypeid"].isna().sum() / total

0.39488452598253326

### latitude
- Latitude of the middle of the parcel multiplied by 10e6
- scalar / 852325
- rate of NaN : 0.3%

In [54]:
print(len(property_2016["latitude"].value_counts()))
property_2016["latitude"].value_counts().head()

852325


33984600.0    862
33708500.0    760
34056700.0    740
34171800.0    611
33766300.0    568
Name: latitude, dtype: int64

In [55]:
property_2016["latitude"].isna().sum() / total

0.0038312122703307665

### longitude
-  Longitude of the middle of the parcel multiplied by 10e6
- scalar / 1042550
- rate of NaN : 0.3%

In [56]:
print(len(property_2016["longitude"].value_counts()))
property_2016["longitude"].value_counts().head()

1042550


-118390000.0    1718
-118438000.0    1585
-118389000.0    1568
-118252000.0    1402
-118432000.0    1371
Name: longitude, dtype: int64

In [57]:
property_2016["longitude"].isna().sum() / total

0.0038312122703307665

### lotsizesquarefeet
- Area of the lot in square feet
- scalar / 69953
- rate of NaN : 9.2%

In [58]:
print(len(property_2016["lotsizesquarefeet"].value_counts()))
property_2016["lotsizesquarefeet"].value_counts().head()

69953


6000.0    44730
5000.0    17546
7200.0    15383
7000.0    11956
6500.0    11624
Name: lotsizesquarefeet, dtype: int64

In [59]:
property_2016["lotsizesquarefeet"].isna().sum() / total

0.09248875374888994

### numberofstories
-  Number of stories or levels the home has
- scalar / 12
- rate of NaN : 77%

In [60]:
print(len(property_2016["numberofstories"].value_counts()))
property_2016["numberofstories"].value_counts().head()

12


1.0    424186
2.0    242125
3.0     15678
4.0        58
5.0        11
Name: numberofstories, dtype: int64

In [61]:
property_2016["numberofstories"].isna().sum() / total

0.7715177824593656

### poolcnt
- Number of pools on the lot (if any)
- scalar / 1
- rate of NaN : 82%

In [62]:
print(len(property_2016["poolcnt"].value_counts()))
property_2016["poolcnt"].value_counts().head()

1


1.0    517534
Name: poolcnt, dtype: int64

In [63]:
property_2016["poolcnt"].isna().sum() / total

0.8266343786733091

### poolsizesum
- Total square footage of all pools on property
- scalar / 907
- rate of NaN : 99%

In [64]:
print(len(property_2016["poolsizesum"].value_counts()))
property_2016["poolsizesum"].value_counts().head()

907


450.0    3179
400.0     972
480.0     917
500.0     828
512.0     826
Name: poolsizesum, dtype: int64

In [65]:
property_2016["poolsizesum"].isna().sum() / total

0.9906338467186807

### pooltypeid10
- Spa or Hot Tub
- category / 1
- rate of NaN : 98.7%

In [66]:
print(len(property_2016["pooltypeid10"].value_counts()))
property_2016["pooltypeid10"].value_counts().head()

1


1.0    36939
Name: pooltypeid10, dtype: int64

In [67]:
property_2016["pooltypeid10"].isna().sum() / total

0.9876260251767291

### pooltypeid2
- Pool with Spa/Hot Tub
- cateory / 1
- rate of NaN : 98.9%

In [68]:
print(len(property_2016["pooltypeid2"].value_counts()))
property_2016["pooltypeid2"].value_counts().head()

1


1.0    32075
Name: pooltypeid2, dtype: int64

In [69]:
property_2016["pooltypeid2"].isna().sum() / total

0.9892553874642949

### pooltypeid7
- Pool without hot tub
- category / 1
- rate of NaN : 83.7%

In [70]:
print(len(property_2016["pooltypeid7"].value_counts()))
property_2016["pooltypeid7"].value_counts().head()

1


1.0    485459
Name: pooltypeid7, dtype: int64

In [71]:
property_2016["pooltypeid7"].isna().sum() / total

0.8373789912090143

### propertycountylandusecode
- County land use code i.e. it's zoning at the county level
- category(object) / 240
- rate of NaN : 0.4%

In [72]:
print(len(property_2016["propertycountylandusecode"].value_counts()))
property_2016["propertycountylandusecode"].value_counts().head()

240


0100    1153896
122      522145
0101     247494
010C     225410
1111     126491
Name: propertycountylandusecode, dtype: int64

In [73]:
property_2016["propertycountylandusecode"].isna().sum() / total

0.004112598849597868

### propertylandusetypeid
- Type of land use the property is zoned for
- category / 15
- rate of NaN : 0.3%

In [74]:
print(len(property_2016["propertylandusetypeid"].value_counts()))
property_2016["propertylandusetypeid"].value_counts().head()

15


261.0    2148362
266.0     481067
246.0     114888
269.0      61469
263.0      58438
Name: propertylandusetypeid, dtype: int64

In [75]:
property_2016["propertylandusetypeid"].isna().sum() / total

0.0038312122703307665

### propertyzoningdesc
- Description of the allowed land uses (zoning) for that property
- category(object) / 5638
- rate of NaN : 33.7%

In [76]:
print(len(property_2016["propertyzoningdesc"].value_counts()))
property_2016["propertyzoningdesc"].value_counts().head()

5638


LAR1     275029
LAR3      67105
LARS      54859
LBR1N     52750
LAR2      48808
Name: propertyzoningdesc, dtype: int64

In [77]:
property_2016["propertyzoningdesc"].isna().sum() / total

0.33719089768013516

### rawcensustractandblock
- Census tract and block ID combined - also contains blockgroup assignment by extension
- category / 99393
- rate of NaN : 0.3%

In [78]:
print(len(property_2016["rawcensustractandblock"].value_counts()))
property_2016["rawcensustractandblock"].value_counts().head()

99393


6.059052e+07    1518
6.037920e+07    1455
6.059063e+07     895
6.037137e+07     886
6.037577e+07     825
Name: rawcensustractandblock, dtype: int64

In [79]:
property_2016["rawcensustractandblock"].isna().sum() / total

0.0038312122703307665

### censustractandblock
- Census tract and block ID combined - also contains blockgroup assignment by extension
- category / 96771
- rate of NaN : 2.5%

In [80]:
print(len(property_2016["censustractandblock"].value_counts()))
property_2016["censustractandblock"].value_counts().head()

96771


6.037920e+13    1171
6.037137e+13     881
6.037920e+13     811
6.059063e+13     807
6.059052e+13     806
Name: censustractandblock, dtype: int64

In [81]:
property_2016["censustractandblock"].isna().sum() / total

0.025166009707167016

### regionidcounty
- County in which the property is located
- category / 3
- rate of NaN : 0.3%

In [82]:
print(len(property_2016["regionidcounty"].value_counts()))
property_2016["regionidcounty"].value_counts().head()

3


3101.0    2009362
1286.0     741565
2061.0     222853
Name: regionidcounty, dtype: int64

In [83]:
property_2016["regionidcounty"].isna().sum() / total

0.0038312122703307665

### regionidcity
- City in which the property is located (if any)
- category / 186
- rate of NaN : 2%

In [84]:
print(len(property_2016["regionidcity"].value_counts()))
property_2016["regionidcity"].value_counts().head()

186


12447.0    670925
46298.0     93470
16764.0     59802
52650.0     58234
25218.0     56962
Name: regionidcity, dtype: int64

In [85]:
property_2016["regionidcity"].isna().sum() / total

0.0210520709214774

### regionidzip
- Zip code in which the property is located
- category / 405
- rate of NaN : 0.4%

In [86]:
print(len(property_2016["regionidzip"].value_counts()))
property_2016["regionidzip"].value_counts().head()

405


96987.0    22021
96193.0    21759
97118.0    20612
97319.0    20310
96964.0    19898
Name: regionidzip, dtype: int64

In [87]:
property_2016["regionidzip"].isna().sum() / total

0.004683076640659624

### regionidneighborhood
- Neighborhood in which the property is located
- category / 528
- rate of NaN : 61%

In [88]:
print(len(property_2016["regionidneighborhood"].value_counts()))
property_2016["regionidneighborhood"].value_counts().head()

528


118208.0    32267
268496.0    23186
48570.0     21186
27080.0     18891
37739.0     18645
Name: regionidneighborhood, dtype: int64

In [89]:
property_2016["regionidneighborhood"].isna().sum() / total

0.6126238059075773

### roomcnt
- Total number of rooms in the principal residence
- scalar / 36
- rate of NaN : 0.3%

In [90]:
print(len(property_2016["roomcnt"].value_counts()))
property_2016["roomcnt"].value_counts().head()

36


0.0    2308879
6.0     178357
7.0     156397
8.0     120227
5.0      99582
Name: roomcnt, dtype: int64

In [91]:
property_2016["roomcnt"].isna().sum() / total

0.0038439416632023737

### storytypeid
- Type of floors in a multi-story house (i.e. basement and main level, split-level, attic, etc.).  See tab for details.
- category / 1
- rate of NaN : 99.9%

In [92]:
print(len(property_2016["storytypeid"].value_counts()))
property_2016["storytypeid"].value_counts().head()

1


7.0    1624
Name: storytypeid, dtype: int64

In [93]:
property_2016["storytypeid"].isna().sum() / total

0.9994559859467502

### typeconstructiontypeid
- What type of construction material was used to construct the home
- category / 5
- rate of NaN : 99.7%

In [94]:
print(len(property_2016["typeconstructiontypeid"].value_counts()))
property_2016["typeconstructiontypeid"].value_counts().head()

5


6.0     6670
4.0       59
13.0      14
10.0       3
11.0       1
Name: typeconstructiontypeid, dtype: int64

In [95]:
property_2016["typeconstructiontypeid"].isna().sum() / total

0.9977398627972439

### unitcnt
-  Number of units the structure is built into (i.e. 2 = duplex, 3 = triplex, etc...)
- category / 146
- rate of NaN : 33%

In [96]:
print(len(property_2016["unitcnt"].value_counts()))
property_2016["unitcnt"].value_counts().head()

146


1.0    1780051
2.0     115652
4.0      39877
3.0      39830
5.0        588
Name: unitcnt, dtype: int64

In [97]:
property_2016["unitcnt"].isna().sum() / total

0.3375724444822604

### yardbuildingsqft17
- Patio in  yard
- scalar / 1636
- rate of NaN : 97.3%

In [98]:
print(len(property_2016["yardbuildingsqft17"].value_counts()))
property_2016["yardbuildingsqft17"].value_counts().head()

1636


240.0    3190
200.0    2980
300.0    2272
180.0    1492
288.0    1443
Name: yardbuildingsqft17, dtype: int64

In [99]:
property_2016["yardbuildingsqft17"].isna().sum() / total

0.9730823588368953

### yardbuildingsqft26
- Storage shed/building in yard
- scalar / 594
- rate of NaN : 99.9%

In [100]:
print(len(property_2016["yardbuildingsqft26"].value_counts()))
property_2016["yardbuildingsqft26"].value_counts().head()

594


120.0    91
100.0    69
240.0    61
160.0    55
80.0     53
Name: yardbuildingsqft26, dtype: int64

In [101]:
property_2016["yardbuildingsqft26"].isna().sum() / total

0.9991132972912857

### yearbuilt
- The Year the principal residence was built 
- category(also use scalar) / 168
- rate of NaN :  2%

In [102]:
print(len(property_2016["yearbuilt"].value_counts()))
property_2016["yearbuilt"].value_counts().head()

168


1955.0    88586
1950.0    76142
1954.0    73307
1953.0    67085
1956.0    65736
Name: yearbuilt, dtype: int64

In [103]:
property_2016["yearbuilt"].isna().sum() / total

0.020074922526570095

### taxvaluedollarcnt
- The total tax assessed value of the parcel
- scalar / 638920
- rate of NaN : 1.4% 

In [104]:
print(len(property_2016["taxvaluedollarcnt"].value_counts()))
property_2016["taxvaluedollarcnt"].value_counts().head()

638920


9.0         1889
400000.0    1572
350000.0    1511
450000.0    1459
500000.0    1339
Name: taxvaluedollarcnt, dtype: int64

In [105]:
property_2016["taxvaluedollarcnt"].isna().sum() / total

0.014253570175970457

### structuretaxvaluedollarcnt
- The assessed value of the built structure on the parcel
- scalar / 426150
- rate of NaN : 1.8%

In [106]:
print(len(property_2016["structuretaxvaluedollarcnt"].value_counts()))
property_2016["structuretaxvaluedollarcnt"].value_counts().head()

426150


100000.0    1880
102461.0    1578
104510.0    1513
200000.0    1418
107401.0    1411
Name: structuretaxvaluedollarcnt, dtype: int64

In [107]:
property_2016["structuretaxvaluedollarcnt"].isna().sum() / total

0.018418091549123564

### landtaxvaluedollarcnt
- The assessed value of the land area of the parcel
- scalar / 531408
- rate of NaN : 2.2%

In [108]:
print(len(property_2016["landtaxvaluedollarcnt"].value_counts()))
property_2016["landtaxvaluedollarcnt"].value_counts().head()

531408


20980.0    3804
22414.0    2474
16275.0    2377
19149.0    2218
9.0        2172
Name: landtaxvaluedollarcnt, dtype: int64

In [109]:
property_2016["landtaxvaluedollarcnt"].isna().sum() / total

0.022689472825593582

### taxamount
- The total property tax assessed for that assessment year
- scalar / 1045757
- rate of NaN : 1%

In [110]:
print(len(property_2016["taxamount"].value_counts()))
property_2016["taxamount"].value_counts().head()

1045757


345.72     171
1844.16    124
1045.88    110
5.02       103
1481.24    102
Name: taxamount, dtype: int64

In [111]:
property_2016["taxamount"].isna().sum() / total

0.010468250716782063

### assessmentyear
- The year of the property tax assessment 
- category / 14
- rate of NaN : 0.3%

In [112]:
print(len(property_2016["assessmentyear"].value_counts()))
property_2016["assessmentyear"].value_counts().head()

14


2015.0    2972420
2014.0       1306
2012.0         26
2004.0          6
2013.0          4
Name: assessmentyear, dtype: int64

In [113]:
property_2016["assessmentyear"].isna().sum() / total

0.0038318822383766405

### taxdelinquencyflag
- Property taxes for this parcel are past due as of 2015
- category / 2 (Y/N)
- rate of NaN : 98%

In [114]:
print(len(property_2016["taxdelinquencyflag"].value_counts()))
property_2016["taxdelinquencyflag"].value_counts().head()

1


Y    56462
Name: taxdelinquencyflag, dtype: int64

In [115]:
property_2016["taxdelinquencyflag"].isna().sum() / total

0.9810861320969296

### taxdelinquencyyear
- Year for which the unpaid propert taxes were due 
- category / 32
- rate of NaN : 98%

In [116]:
print(len(property_2016["taxdelinquencyyear"].value_counts()))
property_2016["taxdelinquencyyear"].value_counts().head()

32


15.0    26068
14.0    16826
13.0     4632
12.0     3124
11.0     2350
Name: taxdelinquencyyear, dtype: int64

In [117]:
property_2016["taxdelinquencyyear"].isna().sum() / total

0.9810854621288838